pkg/controller/apimanager/upgrade_26_to_27.go

```diff
@@ -12,7 +16,13 @@ type Upgrade26_to_27 struct {
 }
 
 func (u *Upgrade26_to_27) Upgrade() (reconcile.Result, error) {
-	res, err := u.upgradeAMPImageStreams()
+
+	res, err := u.upgradeSystemAppPreHookPodCommand()
```

Comment:
We execute the upgrade of the prehook pod command BEFORE the upgrade of the system image.
When the prehook pod command is executed, the prehook pod is NOT redeployed at that moment.

The upgradeSystemAppPreHookPodCommand tries to get the existing system-app DeploymentConfig and checks whether the existing command matches the desired one. When there's a difference the DC is updated and a requeue is performed. This means that at the next iteration of the reconciliation loop we ensure the change is visible when getting it again.

Finally, we upgrade AMP Images after updating the system-app prehook pod. That changes the ImageStream that triggers a DeploymentConfig redeploy, triggering the pre-hook pod execution.

Reply:
`When the prehook pod command is updated, the prehook pod is NOT redeployed at that moment.` I do not understand. We have a trigger of type `configchange`. Then, I would expect that when you change the DC (pre-hook command is changing the DC), a new deployment would be triggered, hence prehook pod will be exected or redeployed.

Reply:
That's not how it works.
The config is changed but the pre-hook pod is only redeployed when the template of the DeploymentConfig changes.

Reply:
right, according to doc: https://docs.openshift.com/container-platform/4.1/applications/deployments/managing-deployment-processes.html#deployments-triggers_deployment-operations
```
The ConfigChange trigger results in a new ReplicationController whenever configuration changes are detected in the Pod template of the DeploymentConfig.
```

Tried myself and changing only prehook command does to trigger creation of a new deployment process